menas/src/main/scala/za/co/absa/enceladus/menas/repositories/OozieRepository.scala

```diff
@@ -0,0 +1,319 @@
+      case Right(client) => fn(client)
+      case Left(ex)      => Future.failed(ex)
+    }
+  }
+
+  private def getOozieClientWrap[T](fn: (OozieClient => T)): Future[T] = {
+    getOozieClient({ cl: OozieClient =>
+      Future(fn(cl))
+    })
+  }
+
+  /**
+   * Get status of submitted coordinater
+   */
+  def getCoordinatorStatus(coordId: String): Future[OozieCoordinatorStauts] = {
+    getOozieClientWrap({ oozieClient: OozieClient =>
+      val jobInfo = oozieClient.getCoordJobInfo(coordId)
+      val nextMaterializeTime = dateFormat.format(jobInfo.getNextMaterializedTime)
+      OozieCoordinatorStauts(jobInfo.getStatus, nextMaterializeTime)
+    })
+  }
+
+  /**
+   * Get status of a running workflow
+   */
+  def getJobStatus(jobId: String): Future[WorkflowStatus] = {
+    getOozieClientWrap({ oozieClient: OozieClient =>
+      oozieClient.getJobInfo(jobId).getStatus
+    })
+  }
+
+  /**
+   * Kill a running coordinator
+   */
+  def killCoordinator(coordId: String): Future[Unit] = {
+    getOozieClientWrap({ oozieClient: OozieClient =>
+      oozieClient.kill(coordId)
+    })
+  }
+
+  /**
+   * Get workflow from teplate - this fills in all variables and returns representation of the workflow
+   */
+  private def getWorkflowFromTemplate(ds: Dataset): Array[Byte] = {
+    val schedule = ds.schedule.get
+    val runtimeParams = schedule.runtimeParams
+    workflowTemplate.replaceAll("\\$stdAppName", s"Menas Schedule Standardization ${ds.name} (${ds.version})")
+      .replaceAll("\\$confAppName", s"Menas Schedule Conformance ${ds.name} (${ds.version})")
+      .replaceAll("\\$stdJarPath", s"$enceladusJarLocation$standardizationJarPath")
+      .replaceAll("\\$confJarPath", s"$enceladusJarLocation$conformanceJarPath")
+      .replaceAll("\\$datasetVersion", schedule.datasetVersion.toString)
+      .replaceAll("\\$datasetName", ds.name)
+      .replaceAll("\\$mappingTablePattern", schedule.mappingTablePattern.flatMap(p => if (p.isEmpty()) None else Some(p)).getOrElse("reportDate={0}-{1}-{2}").trim)
+      .replaceAll("\\$dataFormat", schedule.rawFormat.name)
+      .replaceAll("\\$otherDFArguments", schedule.rawFormat.getArguments.map(arg => s"<arg>$arg</arg>").mkString("\n"))
+      .replaceAll("\\$jobTracker", resourceManager)
+      .replaceAll("\\$sharelibForSpark", oozieShareLib)
+      .replaceAll("\\$nameNode", namenode)
+      .replaceAll("\\$menasRestURI", menasApiURL)
+      .replaceAll("\\$splineMongoURL", splineMongoURL)
+      .replaceAll("\\$stdNumExecutors", runtimeParams.stdNumExecutors.toString)
+      .replaceAll("\\$stdExecutorMemory", s"${runtimeParams.stdExecutorMemory}g")
+      .replaceAll("\\$confNumExecutors", runtimeParams.confNumExecutors.toString)
+      .replaceAll("\\$confExecutorMemory", s"${runtimeParams.confExecutorMemory}g")
+      .replaceAll("\\$driverCores", s"${runtimeParams.driverCores}")
+      .replaceAll("\\$menasCredentialsFile", s"$namenode${runtimeParams.menasCredentialFile}")
```

Comment:
An interesting approach in a positive way

Reply:
Yeah... There's templating engines out there, but none seemed to do exactly what I wanted so this was a simple alternative :)